apps/src/lib/node/ledger/protocol/transactions/store.rs

```diff
@@ -0,0 +1,145 @@
+//! Interfaces for interacting with blockchain state as part of applying a
+//! protocol transaction.
+
+use std::collections::BTreeSet;
+
+use eyre::Result;
+use namada::ledger;
+use namada::ledger::pos::types::WeightedValidator;
+use namada::ledger::storage::{DBIter, StorageHasher, DB};
+use namada::types::address::Address;
+use namada::types::storage::{self, Epoch};
+
+use crate::node::ledger::shell::queries::QueriesExt;
+
+/// Storage functionality needed for applying state changes required by a
+/// protocol transaction. We don't need to know about the gas cost of changes or
+/// the like as gas is not charged for protocol transactions.
+pub(crate) trait Store {
+    /// Returns some value stored at `key`, or `None` if no value is stored
+    /// there.
+    fn read(&self, key: &storage::Key) -> Result<Option<Vec<u8>>>;
+    /// Check if the given key is present in storage.
+    fn has_key(&self, key: &storage::Key) -> Result<bool>;
+    /// Write a value to `key`
+    fn write(
+        &mut self,
+        key: &storage::Key,
+        value: impl AsRef<[u8]> + Clone,
+    ) -> Result<()>;
+}
```

Comment:
tomas is adding a `StoreRead` and `StoreWrite` set of traits, soon, I think

Reply:
These two issues?
https://github.com/anoma/namada/issues/166

Reply:
likewise, add a `TODO`, so we remember to change this later